dev/com.ibm.ws.microprofile.config.1.3/src/com/ibm/ws/microprofile/config13/sources/OSGiConfigUtils.java

```diff
@@ -139,15 +142,22 @@ public static ConfigVariables getConfigVariables(BundleContext bundleContext) {
      * @param bundleContext The context to use to find the service
      * @param serviceClass  The class of the required service
      * @return the service instance or null
+     * @throws InvalidFrameworkStateException
      */
-    public static <T> T getService(BundleContext bundleContext, Class<T> serviceClass) {
+    private static <T> T getService(BundleContext bundleContext, Class<T> serviceClass) throws InvalidFrameworkStateException {
         T service = null;
         if (FrameworkState.isValid()) {
             ServiceReference<T> ref = bundleContext.getServiceReference(serviceClass);
 
             if (ref != null) {
                 service = bundleContext.getService(ref);
             }
+            if (service == null) {
+                String msg = Tr.formatMessage(tc, "service.not.found.CWMCG0204E", serviceClass.getName());
+                throw new ServiceNotFoundException(msg);
```

Comment:
Maybe change this to put the `formatMethod()` in the exception class

Reply:
done